catalog/plugin/catalog-plugin-metacardbackup-s3storage/src/main/java/org/codice/ddf/catalog/plugin/metacard/backup/storage/s3storage/MetacardS3StorageRoute.java

```diff
@@ -0,0 +1,244 @@
+    public String getS3SecretKey() {
+        return s3SecretKey;
+    }
+
+    public void setS3SecretKey(String s3SecretKey) {
+        this.s3SecretKey = s3SecretKey;
+    }
+
+    public String getS3Endpoint() {
+        return s3Endpoint;
+    }
+
+    public void setS3Endpoint(String s3Endpoint) {
+        this.s3Endpoint = s3Endpoint;
+    }
+
+    public String getS3CannedAclName() {
+        return s3CannedAclName;
+    }
+
+    public void setS3CannedAclName(String s3CannedAclName) {
+        this.s3CannedAclName = s3CannedAclName;
+    }
+
+    @Override
+    public void configure() throws Exception {
+        StringBuilder options = new StringBuilder();
+
+        if (StringUtils.isNotBlank(s3AccessKey)) {
+            addQuotedOption(options, AWS_S3_ACCESS_KEY_PROP, s3AccessKey);
+        }
+        if (StringUtils.isNotBlank(s3SecretKey)) {
+            addQuotedOption(options, AWS_S3_SECRET_KEY_PROP, "RAW(" + s3SecretKey + ")");
+        }
+
+        if (options.length() == 0) {
+            AmazonS3Client s3Client = new AmazonS3Client();
+            registry.put("s3Client", s3Client);
+            addOption(options, AWS_S3_CLIENT_PROP, "#s3Client");
+        }
+
+        addOption(options, AWS_S3_ENDPOINT_PROP, s3Endpoint);
+        addOption(options, AWS_S3_DELETE_AFTER_WRITE_PROP, "false");
+
+        String s3Uri = "aws-s3://" + getS3Bucket();
+        if (options.length() > 0) {
+            s3Uri += "?" + options.toString();
+        }
+
+        LOGGER.trace("S3 storage URI: {}", s3Uri);
+
+        String metacardRouteId = "metacard-" + UUID.randomUUID()
+                .toString();
+        from("catalog:postingest").split(method(ResponseMetacardActionSplitter.class,
+                "split(${body})"))
+                .to("direct:" + metacardRouteId);
+        from("direct:" + metacardRouteId).setHeader(METACARD_TRANSFORMER_ID_RTE_PROP,
+                simple(metacardTransformerId, String.class))
+                .setHeader(METACARD_BACKUP_INVALID_RTE_PROP,
+                        simple(String.valueOf(backupInvalidMetacards), Boolean.class))
+                .setHeader(METACARD_BACKUP_KEEP_DELETED_RTE_PROP,
+                        simple(String.valueOf(keepDeletedMetacards), Boolean.class))
+                .choice()
+                .when(not(getCheckDeletePredicate()))
+                .stop()
+                .otherwise()
+                .choice()
+                .when(not(getShouldBackupPredicate()))
+                .stop()
+                .otherwise()
+                .setHeader(S3Constants.KEY,
+                        method(new MetacardTemplateBean(objectTemplate), "applyTemplate(${body})"))
+                .to("catalog:metacardtransformer")
+                .setHeader(S3Constants.CANNED_ACL, simple(s3CannedAclName))
+                .setHeader(S3Constants.CONTENT_LENGTH, simple("${body.length}"))
+                .to(s3Uri);
+
+        LOGGER.trace("Starting metacard S3 storage route: {}", toString());
```

Comment:
I think this is fine. We may decide to turn this rule off since SLF4J is the logger of choice and is still performant here.